src/screens/tabs/Tokens/components/AddToken/steps/CustomToken/index.tsx

```diff
@@ -0,0 +1,201 @@
+import { t } from 'i18next';
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+import { Keyboard, Linking, View } from 'react-native';
+import { Modalize } from 'react-native-modalize';
+import { useDispatch } from 'react-redux';
+
+import Button from '@/components/buttons/Button';
+import RainbowButton from '@/components/buttons/RainbowButton';
+import {
+  ActionButton,
+  ActionSheet,
+  Header,
+  Modal,
+  Text,
+  TextInput,
+} from '@/components/common';
+import ErrorIcon from '@/components/icons/svg/Error.svg';
+import Info from '@/components/icons/svg/Info.svg';
+import { customTokensUrl } from '@/constants/urls';
+import { DABToken } from '@/interfaces/dab';
+import { FungibleStandard } from '@/interfaces/keyring';
+import { getTokenInfo } from '@/redux/slices/user';
+import { validateCanisterId } from '@/utils/ids';
+
+import styles, { iconColor } from './styles';
+import { Error, getError } from './utils';
+
+interface Props {
+  modalRef: React.RefObject<Modalize>;
+  onSelectedToken: (token: DABToken) => void;
+}
+
+function CustomToken({ modalRef, onSelectedToken }: Props) {
+  const [canisterId, setCanisterId] = useState('');
+  const dispatch = useDispatch();
+  const [standard, setStandard] = useState<FungibleStandard>('DIP20');
+  const [loading, setLoading] = useState(false);
+  const [canisterIdError, setCanisterIdError] = useState(false);
+  const [tokenError, setTokenError] = useState<Error>();
+  const optionsRef = useRef<Modalize>(null);
+  const showCanisterError =
+    tokenError?.type === 'INVALID_CANISTER_ID' ||
+    tokenError?.type === 'INVALID_NFT';
+  const showStandardError =
+    tokenError?.type === 'INVALID_STANTARD' || tokenError?.type === 'DEFAULT';
```

Comment:
enum of the types?

Reply:
they are defined as types
![Screen Shot 2022-08-22 at 12 38 10](https://user-images.githubusercontent.com/25931366/185961560-caefb23e-541c-42d1-af2f-df401ec93414.png)
![Screen Shot 2022-08-22 at 12 38 15](https://user-images.githubusercontent.com/25931366/185961570-9d14ce4c-969f-4425-861e-44798fb52e2f.png)